components/browser/state/src/main/java/mozilla/components/browser/state/reducer/LocaleStateReducer.kt

```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.reducer
+
+import mozilla.components.browser.state.action.LocaleAction
+import mozilla.components.browser.state.state.BrowserState
+
+internal object LocaleStateReducer {
+
+    /**
+     * [LocaleAction] Reducer function for modifying [BrowserState.locale].
+     */
+    fun reduce(state: BrowserState, action: LocaleAction): BrowserState {
+        return when (action) {
+            is LocaleAction.UpdateLocaleAction -> state.copy(locale = action.locale)
+            is LocaleAction.RestoreLocaleStateAction -> updateLocaleState(state)
+        }
+    }
+
+    private fun updateLocaleState(state: BrowserState) {
+        state.copy(locale = ...
+            // should we look at the restored app settings here?
+            // request locale from the device? - probably not, unless we're following device lang
+        )
```

Comment:
I created the `restore` functionality that you suggested [in your comment](https://github.com/mozilla-mobile/fenix/pull/18179#issuecomment-786902766) on the fenix PR, but I am unsure what this final step should be 🤔

Reply:
Normally this action doesn't change the state directly but it's used to indicated to listeners that the restore should happen, for example in downloads we use the action to restore downloads [from the db and put them in the browser store](https://github.com/mozilla-mobile/android-components/blob/c5e9046597f4fa8c6d09ac6eff260040b68000af/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadMiddleware.kt#L60), we are doing that work on the `DownloadMiddleware` you could create a `LocaleMiddleware` to perform the restore.